src/chronify/sqlalchemy/functions.py

```diff
@@ -67,8 +69,9 @@ def write_database(
                     raise InvalidOperation(msg)
             conn._dbapi_connection.driver_connection.sql(query)
         case "sqlite":
-            if isinstance(config, DatetimeRange):
-                df = _convert_database_input_for_datetime(df, config)
+            for config in configs:
+                if isinstance(config, DatetimeRange):
+                    df = _convert_database_input_for_datetime(df, config)
```

Comment:
I like that this function takes DatetimeRange specifically and operates on a single column.
Instead of a df.copy(), I am adding and dropping the column that requires formatting.

Reply:
Refer to line 88.

Reply:
I know, I am saying I am changing that. You'll see.

Reply:
I think that the new code is doing this:
1. `df.drop`: Make a copy of the original dataframe without the time column.
2. `pd.concat`: Copy the dataframe from 1 and `dfs` into a new dataframe.
3. `[columns]`: Copy the dataframe from 2 with the columns reordered.

This is three copies per call. Is this correct? There is likely an easy way to reorder columns in place to avoid the third copy.

Reply:
Here is a solution with one copy.
```python
        case "sqlite":
            _check_one_config_per_datetime_column(configs)
            copied = False
            for config in configs:
                if isinstance(config, DatetimeRange):
                    df, copied = _convert_database_input_for_datetime(df, config, copied)
```
```python
def _convert_database_input_for_datetime(
    df: pd.DataFrame, config: DatetimeRange, copied: bool
) -> tuple[pd.DataFrame, bool]:
    if config.is_time_zone_naive():
        return df, copied

    if copied:
        df2 = df
    else:
        df2 = df.copy()
        copied = True

    if isinstance(df2[config.time_column].dtype, DatetimeTZDtype):
        df2[config.time_column] = df2[config.time_column].dt.tz_convert("UTC")
    else:
        df2[config.time_column] = df2[config.time_column].dt.tz_localize("UTC")

    return df2, copied
```

Reply:
Thanks do you just wanna commit that yourself?